lib/shoryuken/queue.rb

```diff
@@ -5,7 +5,11 @@ class Queue
     def initialize(client, name)
       self.name   = name
       self.client = client
-      self.url    = client.get_queue_url(queue_name: name).queue_url
+      begin
+        self.url = client.get_queue_url(queue_name: name).queue_url
+      rescue Aws::SQS::Errors::NonExistentQueue
+        raise StandardError.new "Non existing queue: #{name}. Please create it in advance in SQS."
```

Comment:
Still, if this is an issue for all was SDK users could we fix the error
output there instead of in shoryuken?

On Sun, Nov 8, 2015, 12:37 PM Mathew Hartley <notifications@github.com>
wrote:

> In lib/shoryuken/queue.rb
> <https://github.com/phstc/shoryuken/pull/152#discussion_r44230086>:
>
> > @@ -5,7 +5,11 @@ class Queue
> >      def initialize(client, name)
> >        self.name   = name
> >        self.client = client
> > -      self.url    = client.get_queue_url(queue_name: name).queue_url
> > +      begin
> > +        self.url = client.get_queue_url(queue_name: name).queue_url
> > +      rescue Aws::SQS::Errors::NonExistentQueue
> > +        raise StandardError.new "Non existing queue: #{name}. Please create it in advance in SQS."
>
> The default error makes no mention to the queue which is failing, which
> could be something unexpected if using ActiveJob or prefixing. This may not
> be the best approach but it's helpful to know where to start looking on
> Amazon when you get this error, hence the queue-name.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/phstc/shoryuken/pull/152/files#r44230086>.
>
-- 
Richard Wilson

Software Developer
FreeRunning Technologies
M: (778) 677-5318
TF: (888) 913-2661